doc/architecture/decisions/0004-upgrade-testing.md

```diff
@@ -0,0 +1,40 @@
+# 3. upgrade testing methodology when releasing
+
+Date: 2020-06-25
+
+## Status
+
+Accepted
+
+## Context
+
+One of the goals of the KubeCF project is to offer a smooth experience when it comes to upgrading Cloud Foundry deployments.
+That means any user operating a KubeCF deployment should:
+
+- be able to easily identify which versions he can upgrade to
+- upgrade using documented procedures
+- have confidence that the exact version upgrade he's doing has been tested by the KubeCF team 
+
+## Decision
+
+- not allowed to have minor releases for version X.y.z if X+1 has been released (only patch releases)
+- for a version X.y.z, the upgrade path must be tested for:
+  - the previous X.*.* release
+  - version max(X-1.*.*)
```

Comment:
We explicitly don't support upgrading from `1.0` to `2.0`, I think it is worth to note it in this document.

Reply:
Maybe with a wording such as "Starting from X.Y" could be enough.

Reply:
Can you please give an example? If `1.0` the the max for the `1.*` series, then `2.0` should upgrade from `1.0`.

Reply:
But on the `2.0` release notes we have:
> 

    WARNING: this major version introduces some changes required to consume Eirini as a native component
    by forcing the usage of kubecf name throughout the deployment resources (pod, namespace, ...)
    and so, upgrading from 1.x to 2.x will not be possible!
    More technical details can be found on #477

Hence, saying "starting from 2.0". Not that it matters much, though.

Reply:
Correct, this ADR can't be applied retroactively.